glean_parser/templates/go_server.jinja2

```diff
@@ -224,52 +237,52 @@ func (e {{ metric|event_type_name}}) is{{ ping|ping_events_type_name }}() {}
 {% endif %}
 
 type {{ ping|ping_type_name }} struct {
-    {% for metric_type, metrics in metrics_by_type.items() %}
-    {% if metric_type != 'event' %}
-    {% for metric in metrics %}
-    {{ metric|metric_argument_name }} {{ metric_type|go_metric_type}} // {{ metric.description|clean_string }}
-    {% endfor %}
-    {% endif %}
-    {% endfor %}
-    {% if metrics_by_type['event'] %}
-    Event {{ ping|ping_events_type_name }} // valid event for this ping
-    {% endif %}
+	{% for metric_type, metrics in metrics_by_type.items() %}
+	{% if metric_type != 'event' %}
+	{% for metric in metrics %}
+	{{ metric|metric_argument_name }} {{ metric_type|go_metric_type}} // {{ metric.description|clean_string }}
+	{% endfor %}
+	{% endif %}
+	{% endfor %}
+	{% if metrics_by_type['event'] %}
+	Event {{ ping|ping_events_type_name }} // valid event for this ping
+	{% endif %}
 }
 
 // Record and submit `{{ ping }}` ping
 func (g GleanEventsLogger) Record{{ ping|ping_type_name }}(
-    requestInfo RequestInfo,
-    params {{ ping|ping_type_name }},
-) {
-    var metrics = metrics{
-    {% for metric_type, metrics in metrics_by_type.items() %}
-    {% if metric_type != 'event' %}
-        "{{ metric_type }}": {
-        {% for metric in metrics %}
-            {% if metric_type == 'datetime' %}
+	requestInfo RequestInfo,
+	params {{ ping|ping_type_name }},
+) error {
+	metrics := metrics{
+	{% for metric_type, metrics in metrics_by_type.items() %}
+	{% if metric_type != 'event' %}
+		"{{ metric_type }}": {
+		{% for metric in metrics %}
+			{% if metric_type == 'datetime' %}
 			"{{ metric|metric_name }}": params.{{ metric|metric_argument_name }}.Format("2006-01-02T15:04:05.000Z"),
-            {% else %}
-            "{{ metric|metric_name }}": params.{{ metric|metric_argument_name }},
-            {% endif %}
-        {% endfor %}
-        },
-    {% endif %}
-    {% endfor %}
-    }
-
-    events := []gleanEvent{}
-    {% if metrics_by_type['event'] %}
-    if params.Event != nil {
+			{% else %}
+			"{{ metric|metric_name }}": params.{{ metric|metric_argument_name }},
+			{% endif %}
+		{% endfor %}
+		},
+	{% endif %}
+	{% endfor %}
+	}
+
+	events := []gleanEvent{}
+	{% if metrics_by_type['event'] %}
+	if params.Event != nil {
 		events = append(events, params.Event.gleanEvent())
 	}
-    {% endif %}
-    g.record("{{ ping }}", requestInfo, metrics, events)
+	{% endif %}
+	return g.record("{{ ping }}", requestInfo, metrics, events)
 }
 
 // Record and submit `{{ ping }}` ping omitting user request info
 func (g GleanEventsLogger) Record{{ ping|ping_type_name}}WithoutUserInfo(
-    params {{ ping|ping_type_name}},
-) {
-    g.Record{{ ping|ping_type_name }}(defaultRequestInfo, params)
+	params {{ ping|ping_type_name}},
+) error {
+	return g.Record{{ ping|ping_type_name }}(defaultRequestInfo, params)
```

Comment:
These two generated methods got their signature changed to return error, and the corresponding return statement